experimental/ir/lower_symbols.go

```diff
@@ -95,59 +96,79 @@ func newOneofSymbol(o Oneof) {
 		fqn:  o.InternedFullName(),
 		data: arena.Untyped(c.arenas.oneofs.Compress(o.raw)),
 	})
-	c.symbols = append(c.symbols, ref[rawSymbol]{ptr: sym})
+	c.exported = append(c.exported, ref[rawSymbol]{ptr: sym})
 }
 
-// buildImportedSymbols builds a symbol table of every symbol in every
-// transitive import. This will contain symbols that are not visible to this
-// file, but visibility can be tested with f.visibleImports.
+// buildImportedSymbols builds a symbol table of every imported symbol.
+//
+// It also enhances the exported symbol table with the exported symbols of each
+// public import.
 func buildImportedSymbols(f File, r *report.Report) {
```

Comment:
Name is a bit misleading since this also updates the exported symbol table to include public imports.